cogs/utils/poetry_toolz.py

```diff
@@ -0,0 +1,181 @@
+from random import choice
+from typing import Dict, List, Tuple
+from re import sub, findall, fullmatch
+from pickle import dump
+
+
+def make_rev_gen_dict(ord_word_list: List[str]) -> Dict[str, List[str]]:
+    """
+    Returns a dict containing as keys words and as values a list of words
+    that preceded instances of the key word.
+    Takes as argument an ordered list of words from some text.
+    """
+    prv_word_map: Dict[str, List[str]] = {}
+    for index, word in enumerate(ord_word_list):
+        if word not in prv_word_map:
+            prv_word_map[word] = []
+        if index > 0:
+            prv_word_map[word].append(ord_word_list[index - 1])
+    return prv_word_map
+
+
+def make_rhyme_dict(word_list: List[str],
+                    end_len: int = 4) -> Dict[str, List[str]]:
+    """
+    Returns a dict containing as keys word endings and as values
+    a list of words that match that ending.
+    Takes as argument a list of words from some text, as well
+    as an optional integer, indicating the length of the end slice.
+    """
+    rhyme_dict: Dict[str, List[str]] = {}
+    for word in word_list:
+        stripped_word: str = sub("[^a-z]", "", word.lower())
+        end_slice: str = stripped_word[end_len:]
+        if end_slice not in rhyme_dict:
+            rhyme_dict[end_slice] = list()
+        if stripped_word not in rhyme_dict[end_slice]:
+            rhyme_dict[end_slice].append(stripped_word)
+    return rhyme_dict
+
+
+def syll_count(word_list: List[str]) -> int:
+    """
+    Returns the total amount of syllables in each word in the list of words
+    supplied as argument.
+    Takes as argument a list of strings (words).
+    """
+    total_syll_count: int = 0
+    for word in word_list:
+        curr_word_syll_count: int = 0
+        stripped_word: str = sub("[^a-z]", "", word.lower())
+        for index, char in enumerate(stripped_word):
+            len_word = len(stripped_word)
+            if char in "eyouioa" and (
+                    not (index > 0 and stripped_word[index - 1] in "euioa")
+            ) and (not (char == "e" and index == len_word - 1)):
+                curr_word_syll_count += 1
+        if curr_word_syll_count == 0:
+            curr_word_syll_count = 1
+        total_syll_count += curr_word_syll_count
```

Comment:
i'd prefer these 3 lines to be code-golfed into 1 line
```python
total_syll_count += max(curr_word_syll_count, 1)
```

Reply:
thats a good golf, thx for the suggestion, added